src/Components/KeyDisplayer.re

```diff
@@ -0,0 +1,130 @@
+/*
+ * KeyDisplayer.re
+ *
+ * State for displaying key-presses in the UI
+ */
+
+open Oni_Core;
+
+module List = Utility.List;
+
+// MODEL
+
+// We group key presses in time together,
+// so they show in the same line
+type group = {
+  id: float, // use time of first keypress as id
+  isExclusive: bool,
+  mutable time: float,
+  mutable keys: list(string),
+};
+
+type t = {
+  isEnabled: bool,
+  groups: list(group),
+};
+
+let initial: t = {groups: [], isEnabled: false};
+
+module Constants = {
+  let duration = 2.5;
+  let maxGroupingInterval = 0.3;
+};
+
+// UPDATE-ish
+
+let remvoeExpired = (time, model) => {
```

Comment:
@glennsl Could a typo in `remvoeExpired`

Reply:
Indeed, good catch!

Reply:
Done.